supervisor/coresys.py

```diff
@@ -549,10 +594,12 @@ def sys_hassos(self) -> HassOS:
         """Return HassOS object."""
         return self.coresys.hassos
 
-    def sys_run_in_executor(self, funct, *args) -> asyncio.Future:
+    def sys_run_in_executor(
+        self, funct: Callable[..., T], *args: Any
+    ) -> Coroutine[Any, Any, T]:
         """Add an job to the executor pool."""
         return self.sys_loop.run_in_executor(None, funct, *args)
 
-    def sys_create_task(self, coroutine) -> asyncio.Task:
+    def sys_create_task(self, coroutine: Coroutine) -> asyncio.Task:
```

Comment:
I means this type work like the Callable one?

Reply:
Basically, yes, `create_task` expects a Coroutine.

Since no arguments are taken by this method, both coroutines and coroutine functions are accepted.

e.g.,:

```py
if asyncio.iscoroutine(coroutine):
   return self.sys_loop.create_task(coroutine)

if asyncio.iscoroutinefunction(coroutine):
   return self.sys_loop.create_task(coroutine)
```

In other cases, it should be run in the executor.